//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// File: splitsecret.proto

syntax = "proto2";

message equation_message {
  // Coefficients.
  repeated int32 coefficients = 1;
  // Sum of product of coefficients times their respective variables.
  optional int32 value = 2;
}

message split_secret_message {
  // Name of secret.
  optional string secret_name = 1;
  // Number of 128 components in over-all secret.
  optional int32 number_of_subsequences_in_secret = 2;
  // Sequence number of this component in overall secret.
  optional int32 sequence_number = 3;
  // Number of shards generated.
  optional int32 number_of_shards_outstanding = 4;
  // Number of shards required to reconstruct secret.
  optional int32 number_of_shards_required = 5;
  // Number of this shard in this component.
  optional int32 shard_number = 6;
  // Number of variables in each equation.
  optional int32 number_of_coefficients = 7;
  // Number of equations in this shard.
  optional int32 number_of_equations_in_shard = 8;
  // Equations.
  repeated equation_message equations = 9;
}

